src/Augurk/Extensions/DbFeatureExtensions.cs

```diff
@@ -45,5 +48,79 @@ public static string GetIdentifier(string product, string group, string title, s
                 return id;
             }
         }
+
+        public static FeatureMatch CreateFeatureMatch(this DbFeature feature, string searchTerm)
+        {
+            var match = new FeatureMatch()
+                {
+                    FeatureName = feature.Title,
+                    ProductName = feature.Product,
+                    GroupName = feature.Group,
+                    Version = feature.Version,
+                    Tags = feature.Tags.ToList()
+                };
+
+            string[] searchTerms = searchTerm.Split(" ", StringSplitOptions.RemoveEmptyEntries);
+
+            string matchingText = FindMatchingText(feature.Description, searchTerms);
+            if(String.IsNullOrEmpty(matchingText) && feature.Background != null)
+            {
+                matchingText = FindMatchingText(feature.Background.ToString(), searchTerms);
+            }
+            var enumerator = feature.Scenarios.GetEnumerator();
+            while(String.IsNullOrEmpty(matchingText) && enumerator.MoveNext())
+            {
+                matchingText = FindMatchingText(enumerator.Current.ToString(), searchTerms);
+            }
+
+            match.MatchingText = matchingText;
+
+            return match;
+        }
+
+        private static string FindMatchingText(string source, string[] searchTerms)
```

Comment:
Misschien moeten we hier nog wel wat commentaar toevoegen zodat we weten wat 't doet over een jaar als we weer een keer tijd hebben ;)